lib/chef/knife/bastion_base.rb

```diff
@@ -26,7 +26,7 @@ def tunnel_pid(local_port, raise_on_closed_port = true)
         proxy_pid = pid_result.stdout.chomp
 
         # Verify tunnel destination
-        bastion_ip_addr = Resolv.getaddress(@bastion_host)
+        bastion_ip_addr = Socket.getaddrinfo(@bastion_host, nil, :INET, :STREAM, Socket::IPPROTO_TCP).flatten[3]
```

Comment:
`flatten` is weird here. You receive an array of arrays, where each one represents one network address. Smashing everything together just does not make sense.

Please use `.first` instead